src/piece_picker.cpp

```diff
@@ -333,6 +333,20 @@ namespace libtorrent::aux {
 		*zero_prio = int(m_downloads[piece_pos::piece_zero_prio].size());
 	}
 
+    void piece_picker::set_sequential_range(piece_index_t const start_piece, piece_index_t const end_piece)
+    {
+		INVARIANT_CHECK;
+		m_reverse_cursor = end_piece == piece_index_t(-1) ? m_piece_map.end_index() : next(end_piece);
```

Comment:
The ending piece argument being -1 is a sentinel or special value to represent the ending piece of the torrent and is not what the member variable reverse cursor is set to. The argument is supplied at a time when the user may not know or care for setting the end range; they simply wish for the media to start properly or to seek. I had already placed these sorts of asserts within the torrent handle source file, when you initially mentioned it. I can do so here as well, if it would not be considered too much duplication.